lib/services/ios-project-service.ts

```diff
@@ -400,6 +437,23 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 		}).future<void>()();
 	}
 
+	private validateStaticLibrary(libraryPath: string): IFuture<void> {
+		return (() => {
+			if (path.extname(libraryPath) !== ".a") {
+				this.$errors.failWithoutHelp("The bundle at %s does not contain valid '.a' extension.", libraryPath);
+			}
+
+			let expectedArchs = ["armv7", "arm64", "i386"];
+			let archsInTheFatFile = this.$childProcess.exec("lipo -i " + libraryPath).wait();
+
+			expectedArchs.forEach(expectedArch => {
+				if (archsInTheFatFile.indexOf(expectedArch) < 0) {
+					this.$errors.failWithoutHelp(`The static library at ${libraryPath} is not build for all required architectures - ${expectedArchs}.`);
```

Comment:
I suggest we go with something more general since the library might be built for some of the required architectures. For example "The static library at ${libraryPath} is not built for one or more of the following required architectures: ${expectedArchs}. The static library must be built for all required architectures."